drydock_provisioner/drydock_client/client.py

```diff
@@ -14,83 +14,213 @@
 import json
 import requests
 
-from .session import DrydockSession
+from drydock_provisioner import error as errors
 
 class DrydockClient(object):
     """"
     A client for the Drydock API
 
-    :param string host: Hostname or IP address of Drydock API
-    :param string port: Port number of Drydock API
-    :param string version: API version to access
-    :param string token: Authentication token to use
-    :param string marker: (optional) External marker to include with requests
+    :param DrydockSession session: A instance of DrydockSession to be used by this client
+    :param string version: Drydock API version to use, default is '1.0'
     """
 
-    def __init__(self, host=None, port=9000, version='1.0', token=None, marker=None):
+    def __init__(self, session, version='1.0'):
+        self.session = session
         self.version = version
-        self.session = DrydockSession(token=token, ext_marker=marker)
-        self.base_url = "http://%s:%d/api/%s/" % (host, port, version)
+        self.api_url = "%s/%s/" % (self.session.base_url, version)
 
-    def send_get(self, api_url, query=None):
+    def __send_get(self, endpoint, query=None):
         """
         Send a GET request to Drydock.
 
-        :param string api_url: The URL string following the hostname and API prefix
+        :param string endpoint: The URL string following the hostname and API prefix
         :param dict query: A dict of k, v pairs to add to the query string
         :return: A requests.Response object
         """
-        resp = requests.get(self.base_url + api_url, params=query)
+        resp = self.session.get(self.api_url + endpoint, params=query)
 
         return resp
 
-    def send_post(self, api_url, query=None, body=None, data=None):
+    def __send_post(self, endpoint, query=None, body=None, data=None):
         """
         Send a POST request to Drydock. If both body and data are specified,
         body will will be used.
 
-        :param string api_url: The URL string following the hostname and API prefix
+        :param string endpoint: The URL string following the hostname and API prefix
         :param dict query: A dict of k, v parameters to add to the query string
         :param string body: A string to use as the request body. Will be treated as raw
         :param data: Something json.dumps(s) can serialize. Result will be used as the request body
```

Comment:
By "something json.dumps(s) can serialize" you mean it's just a `dict` right? I think just using ":param dict data:" would be cleaner/easier to read.